File: 2st/Riyeon/최대공약수와 최소공배수.java
class Solution {
    public int[] solution(int n, int m) {
        int[] answer = new int[2];
        int b, s, i=1;
        if(n > m){
            b = n; s = m;
        }else{
            b = m; s = n;
        }
        
        while(answer[1]==0){
            if(n%i==0 && m%i==0 && i<=b)
                answer[0] = i;
            if(i>=s && i%n==0 && i%m==0)
                answer[1] = i;
            i++;
        }
        return answer;
    }
}

File: 2st/Yeonwoo/최대공약수와 최소공배수.py
def solution(n, m):
    answer = []
    temp1 = []
    temp2 = []
    temp3 = []
    for i in range(1,n+1):
        if n%i==0:
            temp1.append(i)
    for i in range(1, m+1):
        if m%i==0:
            temp2.append(i)
    if min(n,m)<=max(n,m):
        for i in temp1:
            for j in temp2:
                if i==j:
                    temp3.append(j)
    temp3.sort(reverse=True)
    answer.append(temp3[0])
    answer.append((n*m)//temp3[0])
    return answer